### DISCUSSION TITLE
Overriding the onRelationManageCreate and onRelationManageUpdate function

### DISCUSSION THREAD

**Discussion by LeonKercmar** [OFFLINE-GmbH/oc-mall-plugin#1074] [Q&A]
Hello,

I have a question regarding overriding the two functions. Under variants I've created four custom fields including a Discount Price field which works the same as the Price field. The problem is if I want it to work I have to override the onRelationManageCreate and onRelationManageUpdate function. Is there a way I can do that? I know I can manually change the code but every time I want to update the plugin my code overrides. 

![image](https://github.com/OFFLINE-GmbH/oc-mall-plugin/assets/105213431/830355d0-f969-4620-8130-31b6022d8905)

Any help would be appreciated.

**Comment by tobias-kuendig** [comment]
You probably just have to add your fields to the fillable property:

https://github.com/OFFLINE-GmbH/oc-mall-plugin/blob/02f50ddcf3c601a6cdd1aadf847ce17a0bb1965c/models/Variant.php#L97

Try doing it manually and see if you data is stored (without any custom code in the controller methods). If it works, you can use the Model::extend hook to patch the fillable property from your own plugin.

**Comment by LeonKercmar** [comment]
Thank you for the response. Adding it to fillable doesn't help.

The Prices field is being saved in the products.php controller 
![image](https://github.com/OFFLINE-GmbH/oc-mall-plugin/assets/105213431/03600009-7fb6-49c6-8d58-c42c04f6cd8b)
and my Custom field is made with the same logic.

**Comment by tobias-kuendig** [comment]
You can also try to execute your custom code in the `beforeSave` Event of the respective model. You have access to the post data there. Just make sure to only run your code in beforeSafe if the app is running in the backend and the relevant ajax handler is run.

We do a check for something similar here:

https://github.com/OFFLINE-GmbH/oc-mall-plugin/blob/02f50ddcf3c601a6cdd1aadf847ce17a0bb1965c/classes/observers/PropertyValueObserver.php#L50
